pyterrier/text.py

```diff
@@ -386,14 +397,32 @@ def __init__(self, **kwargs):
 
 class SlidingWindowPassager(Transformer):
 
-    def __init__(self, text_attr='body', title_attr='title', passage_length=150, passage_stride=75, join=' ', prepend_title=True, **kwargs):
+    def __init__(self, text_attr='body', title_attr='title', passage_length=150, passage_stride=75, join=' ', prepend_title=True, tokenizer=None, **kwargs):
         super().__init__(**kwargs)
         self.text_attr=text_attr
         self.title_attr=title_attr
         self.passage_length = passage_length
         self.passage_stride= passage_stride
         self.join = join
         self.prepend_title = prepend_title
+        self.tokenizer = tokenizer
+
+        # check if the tokenizer has the `.tokenize()` and the `.convert_tokens_to_string()` method
+        if self.tokenizer is not None:
+            tokenizer_attr = getattr(self.tokenizer, 'tokenize', False)
+            if not tokenizer_attr or not callable(tokenizer_attr):
```

Comment:
not sure why being callable is relevant, given we expect it to have tokenize() and convert_tokens_to_string() are expected.

Reply:
as mentioned, `.tokenize()` and `.convert_tokens_to_string()` are expected, so both these checks are to enforce this requirement and make sure they are functions and throw an error if they aren't. we can remove it since it'll throw an error anyways when being called